### PR TITLE
[WIP] Implement fallback server functionality in SocketService

### PR DESCRIPTION
**Pull request by cameronaaron**
Introduce a secondary server URL for socket connections, allowing for automatic switching between primary and fallback servers in case of connection issues. Update relevant configurations and settings to support this feature.

### CODE REVIEW

File: android/.project
Comment (zlshames) [inline android/.project:1]: You added these to the gitignore, but still included them in your commit

File: android/.settings/org.eclipse.buildship.core.prefs
Comment (zlshames) [inline android/.settings/org.eclipse.buildship.core.prefs:1]: You added these to the gitignore, but still included them in your commit

File: lib/services/network/socket_service.dart [hunk 1]
```diff
@@ -155,39 +164,35 @@ class SocketService extends GetxService {
...
         Logger.info("Disconnected from socket...");
         state.value = SocketState.disconnected;
-        return;
+        break;
       case SocketState.connecting:
         Logger.info("Connecting to socket...");
         state.value = SocketState.connecting;
-        return;
+        break;
       case SocketState.error:
```
Comment (zlshames) [inline lib/services/network/socket_service.dart:176]: I'm not sure i liked that we changed the functionality of fetching from firebase. Especially when not everyone will have a secondary address entered.

Can we maintain the old functionality based on whether or not there is a secondary server address set? If not set, fetch new url from firebase. I also think that if both server URLs are tried before it works, it should also fetch the new url from firebase. Essentially, keep a counter of errors/failures. It's incremented on error, and reset on connection. If errors == serverAddresses.length, then also try fetching new url from firebase

**Review by zlshames — CHANGES_REQUESTED** [review]

**Comment by zlshames** [comment]
Also, @cameronaaron i assume that you didn't implement any settings UI to configure this secondary URL? Just the code to handle it?

**Comment by cameronaaron** [comment]
Exactly this is all a work in progress! I just started it and made the base pull request so that I will actually finish it lol

**Comment by zlshames** [comment]
ok, in that case, i'll hold off on merging it for this next release

File: lib/services/network/socket_service.dart (hunk 1 above)
Comment (cameronaaron) [reply lib/services/network/socket_service.dart:176]: > I'm not sure i liked that we changed the functionality of fetching from firebase. Especially when not everyone will have a secondary address entered.
> 
> Can we maintain the old functionality based on whether or not there is a secondary server address set? If not set, fetch new url from firebase. I also think that if both server URLs are tried before it works, it should also fetch the new url from firebase. Essentially, keep a counter of errors/failures. It's incremented on error, and reset on connection. If errors == serverAddresses.length, then also try fetching new url from firebase



Ah yes I agree that is a better method will do that

**Pull request closed without merging**